client/blocks/login/index.jsx

```diff
@@ -337,413 +328,20 @@ class Login extends Component {
-					headerText = translate( 'Log in to WP Cloud with WordPress.com' );
-					preHeader = (
-						<div>
-							<WPCloudLogo className="login__wpcloud-logo" size={ 256 } />
-						</div>
-					);
-				} else if ( document.location.search?.includes( 'jetpack' ) ) {
-					headerText = translate(
-						'Howdy! Log into the Jetpack Partner Portal with your WordPress.com account.'
-					);
-					preHeader = (
-						<div>
-							<JetpackPlusWpComLogo className="login__jetpack-plus-wpcom-logo" size={ 24 } />
-						</div>
-					);
-				} else {
-					headerText = translate(
-						'Howdy! Log into the Automattic Partner Portal with your WordPress.com account.'
-					);
-				}
-			}
-
-			if ( isJetpackCloudOAuth2Client( oauth2Client ) || isA4AOAuth2Client( oauth2Client ) ) {
-				// If users arrived here from the lost password flow, show them a specific message about it
-				postHeader = currentQuery.lostpassword_flow && (
-					<p className="login__form-post-header">
-						{ translate(
-							'Check your e-mail address linked to the account for the confirmation link, including the spam or junk folder.'
-						) }
-					</p>
-				);
-			}
-
-			if ( isCrowdsignalOAuth2Client( oauth2Client ) ) {
-				headerText = translate( 'Sign in to %(clientTitle)s', {
-					args: {
-						clientTitle: oauth2Client.title,
-					},
-				} );
-			}
-
-			if ( isGravPoweredClient ) {
-				headerText = translate( 'Login to %(clientTitle)s', {
-					args: { clientTitle: oauth2Client.title },
-				} );
-
-				if ( isGravPoweredLoginPage ) {
-					const isFromGravatar3rdPartyApp =
-						isGravatarOAuth2Client( oauth2Client ) && currentQuery?.gravatar_from === '3rd-party';
-					const isFromGravatarQuickEditor =
-						isGravatarOAuth2Client( oauth2Client ) &&
-						currentQuery?.gravatar_from === 'quick-editor';
-					const isGravatarFlowWithEmail = !! (
-						isGravatarFlowOAuth2Client( oauth2Client ) && currentQuery?.email_address
-					);
-
-					postHeader = (
-						<p className="login__header-subtitle">
-							{ isFromGravatar3rdPartyApp || isFromGravatarQuickEditor || isGravatarFlowWithEmail
-								? translate( 'Please log in with your email and password.' )
-								: translate(
-										'If you prefer logging in with a password, or a social media account, choose below:'
-								  ) }
-						</p>
-					);
-				}
-			}
-
-			if ( isBlazeProOAuth2Client( oauth2Client ) ) {
-				headerText = <h3>{ translate( 'Log in to your Blaze Pro account' ) }</h3>;
-
-				postHeader = (
-					<p className="login__header-subtitle">
-						{ translate( "Don't have an account? {{signupLink}}Sign up here{{/signupLink}}", {
-							components: { signupLink },
-						} ) }
-					</p>
-				);
-
-				if ( this.showContinueAsUser() ) {
-					postHeader = (
-						<p className="login__header-subtitle">
-							{ translate( 'Select the account you’d like to use' ) }
-						</p>
-					);
-				}
-			}
-		} else if ( isWooJPC ) {
```

Comment:
When the flow is `isWooJPC` then `isWhiteLogin` is false and this `LoginHeader` component is not even rendered. That's one reason why this flow (including "lost password") has incorrect header and sub-header.